az-reportal/src/main/java/azkaban/viewer/reportal/ReportalServlet.java

```diff
@@ -1045,7 +1045,19 @@ private String validateAndSaveReport(final HttpServletRequest req,
     page.add("projectId", project.getId());
 
     try {
-      report.createZipAndUpload(projectManager, user, this.reportalStorageUser);
+      // Fetch the uploader's IP
+      String uploaderIPAddr = "";
+      if (req != null) {
+        uploaderIPAddr = req.getHeader("X-FORWARDED-FOR");
+        if (uploaderIPAddr == null || "".equals(uploaderIPAddr)) {
+          logger.info(" Failed to fetch remote Address using \"X-FORWARDED-FOR\"");
```

Comment:
Do you feel these are all INFO level messages, or more like DEBUG? Also, any special meaning the the leading space?

Reply:
The 1st one can be debug. The rest should be info. This is logged only once per upload.